src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs

```diff
@@ -88,6 +88,28 @@ public class GenevaExporterOptions
             var copy = new Dictionary<string, object>(value.Count + 1) { [Schema.V40.PartA.Ver] = schemaVersion };
             foreach (var entry in value)
             {
+                var val = entry.Value;
+                
+                switch (val)
+                {
+                    case bool vb:
+                    case byte vui8:
+                    case sbyte vi8:
+                    case short vi16:
+                    case ushort vui16:
+                    case int vi32:
+                    case uint vui32:
+                    case long vi64:
+                    case ulong vui64:
+                    case float vf:
+                    case double vd:
+                    case string vs:
+                        break;
+                    default:
+                        val = val.ToString();
```

Comment:
I suggest taking a different approach here by explicitly asking the user to do the string conversion. Similar to what @utpilla explained in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/395#issue-1262454425.

Main reason of not using `ToString` - for arbitrary types, this would introduce localization/globalization issue (e.g. `val` might be converted to different string depending on the user language/locale preference). We don't know if the user's intention is to have an InvariantCulture string or not, only the user knows what they want.

In a nutshell, I think the spirit here should be giving a string saying "hey, you've provided a non-primitive type which we do not support, you probably want to convert it to a string (invariant or local culture - you decide)?".

Reply:
@reyang  Thank you so much! I have updated the code, can you help review again?